### PR TITLE
Merge in appliance features

### PR DESCRIPTION
**Pull request by ameliang96**
Merge in appliance features
-List appliances (I got rid of follow ups)
-Delete an appliance
-Create an appliance
-Get info on an appliance

### CODE REVIEW

File: actions/create_appliance.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 from requests.auth import HTTPBasicAuth
 import requests
-# import app
-
+from util import get_site_id_by_name
 
 def create_appliance(api_auth, parameters, contexts):
     """
+    Allows users to create an appliance on a site. 
+    We in order for them to do so, we need to know the city, site name
```
Comment (mirrexagon) [inline actions/create_appliance.py]: Typo? Looks like it should just be "In order for [...]"

File: actions/create_appliance.py [hunk 2]
```diff
@@ -16,32 +18,32 @@ def create_appliance(api_auth, parameters, contexts):
...
+        country_code = parameters["Country"]["alpha-2"]
+ 
     except KeyError as e:
         error_string = "Error processing create Appliance intent. {0}".format(e)
         logging.error(error_string)
         return error_string
 
     # Get all sites and check whether site exists
+    # Currently, if site doesn't exist it just says not available
+    # Need to fix it, but depends on how get_site_id_by_name is impleented
```
Comment (mirrexagon) [inline actions/create_appliance.py]: More typo, "impleented"

File: actions/delete_appliance_followup.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    :type api_auth: SteelConnectAPI
+    :param parameters: json parameters from Dialogflow intent
+    :type parameters: json
+    :return: Returns a response to be read out to user
+    :rtype: string
+    """
+    try:
+        option_choice = int(parameters["OptionNumber"])
+    except KeyError as e:
+        error_string = "Error processing deleteing appliance followup"
```
Comment (mirrexagon) [inline actions/delete_appliance_followup.py]: Typo "deleteing", should be "deleting".

File: actions/get_appliance_info.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import logging
+import requests
+from flask import json
+from requests.auth import HTTPBasicAuth
+from util import get_site_id_by_name
+
+def get_appliance_info(api_auth, parameters, contexts):
+    """
+    Allow users to get information about a oarticular appliance
```
Comment (mirrexagon) [inline actions/get_appliance_info.py]: "oarticular"

File: actions/create_appliance.py [hunk 5]
```diff
@@ -16,32 +18,32 @@ def create_appliance(api_auth, parameters, contexts):
...
+ 
     except KeyError as e:
         error_string = "Error processing create Appliance intent. {0}".format(e)
         logging.error(error_string)
         return error_string
 
     # Get all sites and check whether site exists
+    # Currently, if site doesn't exist it just says not available
+    # Need to fix it, but depends on how get_site_id_by_name is impleented 
+    site_id = get_site_id_by_name(api_auth, site_name, city,country_code)
```
Comment (mirrexagon) [inline actions/create_appliance.py]: Should this catch util.APIError exceptions and return the error string if there is one? ie.

```python
from util import APIError

[...]

try:
    site_id = get_site_id_by_name(api_auth, site_name, city,country_code)
except APIError as e:
    return str(e)
```

File: actions/delete_appliance.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        site_name = parameters["SiteName"]
+        model = parameters["Model"]
+        country_code = parameters["Country"]["alpha-2"]
+ 
+    except KeyError as e:
+        error_string = "Error processing getting Appliance information intent. {0}".format(e)
+        logging.error(error_string)
+        return error_string
+
+    site_id = get_site_id_by_name(api_auth, site_name, city,country_code)
```
Comment (mirrexagon) [inline actions/delete_appliance.py]: As above, should this catch APIError?

**Comment by mirrexagon** [comment]
Oh, one thing I forgot to mention, you might want to include your Dialogflow changes. What I did was export the agent as zip, then copy the intents I changed/added from `intents/` inside the zip to `agents/intents/` in this repo. I didn't test reimporting that though...

**Comment by ameliang96** [comment]
Cool. Done!

**Comment by mirrexagon** [comment]
Looks good!

**Pull request merged**